File: query_users.py
import sqlite3

# 连接到数据库
conn = sqlite3.connect('./backend/database/education.db')
cursor = conn.cursor()

# 查询用户数量
cursor.execute("SELECT COUNT(*) FROM users")
count = cursor.fetchone()[0]
print(f"用户总数: {count}\n")

# 查询表结构
cursor.execute("PRAGMA table_info(users)")
columns = cursor.fetchall()
print("表结构:")
for col in columns:
    print(f"  {col[1]} ({col[2]})")
print()

# 查询用户列表 - 使用实际存在的列
cursor.execute("SELECT id, username, email, role FROM users ORDER BY id")
users = cursor.fetchall()

print(f"{'ID':<5} {'用户名':<20} {'邮箱':<30} {'角色':<10}")
print("--------------------------------------------------------------------")

for user in users:
    id, username, email, role = user
    # 处理可能的None值
    email = email if email else ""
    print(f"{id:<5} {username:<20} {email:<30} {role:<10}")

# 关闭连接
conn.close()

File: query_users.go
package main

import (
	"database/sql"
	"fmt"
	"log"

	_ "github.com/mattn/go-sqlite3"
)

func main() {
	// 数据库连接
	db, err := sql.Open("sqlite3", "./backend/database/education.db")
	if err != nil {
		log.Fatal(err)
	}
	defer db.Close()

	// 查询用户数量
	var count int
	err = db.QueryRow("SELECT COUNT(*) FROM users").Scan(&count)
	if err != nil {
		log.Fatal(err)
	}

	fmt.Printf("用户总数: %d\n\n", count)

	// 查询用户列表
	rows, err := db.Query("SELECT id, username, full_name, role FROM users ORDER BY id")
	if err != nil {
		log.Fatal(err)
	}
	defer rows.Close()

	fmt.Printf("%-5s %-20s %-30s %-10s\n", "ID", "用户名", "姓名", "角色")
	fmt.Println("--------------------------------------------------------------------")

	for rows.Next() {
		var id int
		var username, fullName, role string
		err := rows.Scan(&id, &username, &fullName, &role)
		if err != nil {
			log.Fatal(err)
		}
		fmt.Printf("%-5d %-20s %-30s %-10s\n", id, username, fullName, role)
	}

	if err = rows.Err(); err != nil {
		log.Fatal(err)
	}
}